### PR TITLE
feat: add compute virtual method to Event Condition

### PR DESCRIPTION
**Pull request by vishwa2710**
- Add a compute method that returns the computed value
- Add a "target" member variable to EventCondition
- evaluate is now a non-virtual parent method in EventCondition that calls compute and subtracts the target

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp [hunk 1]
```diff
@@ -36,12 +37,23 @@ EventCondition::Criteria EventCondition::getCriteria() const
...
 }
 
+Real EventCondition::getTarget() const
+{
+    return target_;
+}
+
+Real EventCondition::evaluate(const VectorXd& aStateVector, const Real& aTime) const
+{
+    return compute(aStateVector, aTime) - target_;
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp:47]: Does the order/sign of the substraction matter for any use case? 🤔

File: src/OpenSpaceToolkit/Astrodynamics/EventCondition/DurationCondition.cpp [hunk 2]
```diff
@@ -10,22 +10,21 @@ namespace eventcondition
...
-    return Duration::Seconds(durationInSeconds_);
+    return Duration::Seconds(getTarget());
 }
 
-Real DurationCondition::evaluate(const VectorXd& aStateVector, const Real& aTime) const
+Real DurationCondition::compute(const VectorXd& aStateVector, const Real& aTime) const
 {
     (void)aStateVector;
-    return aTime - durationInSeconds_;
+    return aTime;
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/EventCondition/DurationCondition.cpp:27]: Alright, should all be backward compatible as `evaluate` remains the method computing the difference between target and value, confirmed?

**Comment by Derollez** [comment]
So `COECondition` already had `target`. Here we are moving that attribute to the base class `EventCondition`, I am assuming this is making our lives easier in general to have `target` defined at `EventCondition` level later on when manipulating them but jut curious if you have an example for this 👀

File: src/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.cpp [hunk 3]
```diff
@@ -53,39 +47,14 @@ COE::Element COECondition::getElement() const
...
-        case COE::Element::TrueAnomaly:
-            return "True anomaly";
-        case COE::Element::MeanAnomaly:
-            return "Mean anomaly";
-        case COE::Element::EccentricAnomaly:
-            return "Eccentric anomaly";
-    }
-
-    throw ostk::core::error::runtime::Wrong("Element");
+    return evaluator_(positionVector, velocityVector, this->gravitationalParameter_);
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.cpp:57]: Should we rename the `evaluator_` and the `GetEvaluator` items to match the difference we are now making between `compute` and `evaluate` methods?

File: src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp (hunk 1 above)
Comment (Derollez) [reply src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp:47]: Alright, LGTM. Checked different target values (positive, negative) at `isSatisfied` level

File: src/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.cpp (hunk 3 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/EventCondition/COECondition.cpp:57]: Yes this is a good idea!

File: bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/EventCondition.cpp [hunk 4]
```diff
@@ -58,11 +58,22 @@ inline void OpenSpaceToolkitAstrodynamicsPy_EventCondition(pybind11::module& aMo
...
+                init<const String&, const EventCondition::Criteria&, const Real&>(),
+                arg("name"),
+                arg("criteria"),
+                arg("target")
+            )
 
             .def("__str__", &(shiftToString<EventCondition>))
             .def("__repr__", &(shiftToString<EventCondition>))
 
+            .def("get_name", &EventCondition::getName)
```
Comment (kyle-cochran) [inline bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/EventCondition.cpp:71]: Not 100% about this MR but would be nice to have some kind of glossary (maybe as part of the doc strings) that say what all of these words mean. 

It's not very obvious what `target` or `criteria` means or what the difference is between `compute` and `evaluate`. Maybe adding a class-level docstring to Event conditions could help?

Comment (vishwa2710) [reply bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/EventCondition.cpp:65]: Discussed this with Kyle and we agreed that the doxygen comments are sufficient for now...

**Review by phc1990 — APPROVED** [review]
Looks great!

Left just a couple of nit comments.

I see benchmark is complaining, lmk if you'd like me to take a look at it : )

File: include/OpenSpaceToolkit/Astrodynamics/EventCondition.hpp [hunk 5]
```diff
@@ -41,7 +41,8 @@ class EventCondition
     /// @endcode
     ///
     /// @param                  [in] aName A string representing the name of the Event Condition
-    /// @param                  [in] aCriteria An enum indicating the criteria used to determine the Event Condition
+    /// @param                  [in] aCriteria An enum indicating the criteria used to determine if the Event Condition
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/EventCondition.hpp:44]: While technically correct I believe Criteria is used as a singular plural, like data. I think it's fine as it does not create ambiguity IMO

**Pull request merged**